spec.md

```diff
@@ -5,22 +5,57 @@ scripts provided by this repository will support one or more versions of this
 file, but they should all be considered internal investigations, not intended
 for public re-use.
 
-## Basic layout
+## On-disk (or in-cloud) layout
 
 ```
+
 .                             # Root folder, potentially in S3,
-├── 123.zarr                  # with a flat list of images by image ID.
-└── 456.zarr                  #
-    ├── .zattrs               # Group level metadata.
-    ├── .zgroup               # Each image is a Zarr group with multscale metadata.
-    └── 0                     # Each multiscale level is stored as a separate Zarr array.
-        ├── .zarray           #
-        ├── 0.0.0.0.0         # Chunks are stored with the flat directory layout.
-        └── t.c.z.y.x         # All image arrays are 5-dimensional
-                              # with dimension order (t, c, z, y, x).
+│                             # with a flat list of images by image ID.
+│
+├── 123.zarr                  # One image (id=123) converted to Zarr.
+│
+└── 456.zarr                  # Another image (id=456) converted to Zarr.
+    │
+    ├── .zgroup               # Each image is a Zarr group, or a folder, of other groups and arrays.
+    ├── .zattrs               # Group level attributes are stored in the .zattrs file and include
+    │                         #  "multiscales" and "omero" below)
+    │
+    ├── 0                     # Each multiscale level is stored as a separate Zarr array,
+    │   ...                   # which is a folder containing chunk files which compose the array.
+    ├── n                     # The name of the array is arbitrary with the ordering defined by
+    │   │                     # by the "multiscales" metadata, but is often a sequence starting at 0.
+    │   │
+    │   ├── .zarray           # All image arrays are 5-dimensional
+    │   │                     # with dimension order (t, c, z, y, x).
+    │   │
+    │   ├── 0.0.0.0.0         # Chunks are stored with the flat directory layout.
+    │   │   ...               # Each dotted component of the chunk file represents
+    │   └── t.c.z.y.x         # a "chunk coordinate", where the maximum coordinate
+    │                         # will be `dimension_size / chunk_size`.
+    │
+    └── masks
+        │
+        ├── .zgroup           # The masks group is a container which holds a list
+        ├── .zattrs           # of masks to make the objects easily discoverable,
+        │                     # All masks will be listed in `.zattrs` e.g. `{ "masks": [ "original/0" ] }`
+        │
+        └── original          # Intermediate folders are permitted but not necessary
+            │                 # and currently contain no extra metadata.
+            └── 0
+                ├── .zgroup   # Each mask itself is also a multiscale image with the extra key
```

Comment:
I don't have a `.zgroup` having exported with `omero zarr masks Image:52` using `https://github.com/ome/omero-cli-zarr/pull/12/commits/313af34d151b824426f7f009dbfce833d2aee25c`.
I see `.zarray` instead in `$ ls -al 52.zarr/masks/0/`

Reply:
My apologies. Yes. One batch of work didn't get opened. Thoughts on either reverting this to say "just an array" for the moment or hold off until we have the sub-resolutions? (This goes back to my question of whether or not we _must_ have matching sub-resolutions)

Reply:
I think for now this should just describe the current state `.zarray` so we can merge this along with ome-zarr-py and omero-cli-zarr PRs. Then think about how to handle multi-resolutions

Reply:
Updated in https://github.com/ome/omero-ms-zarr/commit/c15b68b9e8e61e7fd7c547af11f90c91509a5aab